packages/react-reconciler/src/ReactFiberHostContext.new.js

```diff
@@ -14,33 +14,26 @@ import type {Container, HostContext} from './ReactFiberHostConfig';
 import {getChildHostContext, getRootHostContext} from './ReactFiberHostConfig';
 import {createCursor, push, pop} from './ReactFiberStack.new';
 
-declare class NoContextT {}
-const NO_CONTEXT: NoContextT = ({}: any);
-
-const contextStackCursor: StackCursor<HostContext | NoContextT> = createCursor(
-  NO_CONTEXT,
-);
-const contextFiberStackCursor: StackCursor<Fiber | NoContextT> = createCursor(
-  NO_CONTEXT,
+const contextStackCursor: StackCursor<HostContext | null> = createCursor(null);
+const contextFiberStackCursor: StackCursor<Fiber | null> = createCursor(null);
+const rootInstanceStackCursor: StackCursor<Container | null> = createCursor(
+  null,
 );
-const rootInstanceStackCursor: StackCursor<
-  Container | NoContextT,
-> = createCursor(NO_CONTEXT);
-
-function requiredContext<Value>(c: Value | NoContextT): Value {
-  if (c === NO_CONTEXT) {
-    throw new Error(
-      'Expected host context to exist. This error is likely caused by a bug ' +
-        'in React. Please file an issue.',
-    );
-  }
 
+function requiredContext<Value>(c: Value | null): Value {
+  if (__DEV__) {
+    if (c === null) {
+      console.error(
+        'Expected host context to exist. This error is likely caused by a bug ' +
+          'in React. Please file an issue.',
+      );
+    }
+  }
   return (c: any);
 }
 
 function getCurrentRootHostContainer(): null | Container {
-  const container = rootInstanceStackCursor.current;
-  return container === NO_CONTEXT ? null : ((container: any): Container);
```

Comment:
This one is precious.

Reply:
I was busy!

Reply:
I mean it's fair, once a pattern is established it tends to spread.